flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/common/OverAggregateTestPrograms.java

```diff
@@ -0,0 +1,433 @@
+        "+I[Hello, 16, 1970-01-01T00:00:16, 16, 0, 3, 3]",
+        "+I[Hello, 20, 1970-01-01T00:00:20, 2, 0, 4, 5]",
+        "+I[Hello, 30, 1970-01-01T00:00:30, 3, 0, 5, 8]",
+        "+I[Hello, 40, 1970-01-01T00:00:40, 4, 0, 6, 12]",
+        "+I[Hello, 50, 1970-01-01T00:00:50, 5, 1, 6, 16]",
+        "+I[Hello, 60, 1970-01-01T00:01, 6, 2, 6, 21]",
+        "+I[Hello, 65, 1970-01-01T00:01:05, 65, 3, 6, 26]",
+        "+I[Hello, 90, 1970-01-01T00:01:30, 9, 4, 6, 30]",
+        "+I[Hello, 95, 1970-01-01T00:01:35, 18, 5, 6, 33]",
+        "+I[Hello World, 100, 1970-01-01T00:01:40, 7, 1, 1, 7]",
+        "+I[Hello, 99, 1970-01-01T00:01:39, 15, 6, 6, 48]",
+        "+I[Hello World, 110, 1970-01-01T00:01:50, 17, 2, 2, 14]",
+        "+I[Hello World, 140, 1970-01-01T00:02:20, 18, 3, 3, 21]",
+        "+I[Hello World, 150, 1970-01-01T00:02:30, 8, 4, 4, 29]",
+        "+I[Hello World, 200, 1970-01-01T00:03:20, 20, 5, 5, 49]"
+    };
+
+    private static final String[] AFTER_RESTORE_DATA_PRECEDING_ROWS = {
+        "+I[Hello World, 202, 1970-01-01T00:03:22, 20, 6, 6, 72]"
+    };
+
+    private static final String[] AFTER_RESTORE_DATA_PRECEDING_ROWS_WITH_OUT_OF_ORDER_RECORDS = {
+        "+I[Hello World, 202, 1970-01-01T00:03:22, 20, 6, 6, 69]"
+    };
+
+    public static final TableTestProgram OVER_AGGREGATE_ROW_BOUNDED_PARTITIONED_PRECEDING_ROWS =
+            getTableTestProgram(
+                    "over-aggregate-bounded-partitioned-preceding-rows",
+                    "validates over aggregate node partitioned and bounded by prior rows",
+                    "PARTITION BY c ORDER BY rowtime ROWS BETWEEN 5 preceding AND CURRENT ROW",
+                    SOURCE,
+                    BEFORE_RESTORE_DATA_PRECEDING_ROWS,
+                    AFTER_RESTORE_DATA_PRECEDING_ROWS);
+
+    public static final TableTestProgram
+            OVER_AGGREGATE_ROW_BOUNDED_PARTITIONED_PRECEDING_ROWS_WITH_OUT_OF_ORDER_RECORDS =
+                    getTableTestProgram(
+                            "over-aggregate-bounded-partitioned-preceding-rows-with-out-of-order-records",
+                            "validates over aggregate node partitioned and bounded by prior rows",
+                            "PARTITION BY c ORDER BY rowtime ROWS BETWEEN 5 preceding AND CURRENT ROW",
+                            SOURCE_WITH_OUT_OF_ORDER_RECORDS,
+                            BEFORE_RESTORE_DATA_PRECEDING_ROWS_WITH_OUT_OF_ORDER_RECORDS,
+                            AFTER_RESTORE_DATA_PRECEDING_ROWS_WITH_OUT_OF_ORDER_RECORDS);
+
+    private static TableTestProgram getTableTestProgram(
+            final String id,
+            final String description,
+            final String windowSql,
+            final SourceTestStep source,
+            final String[] beforeRows,
+            final String[] afterRows) {
+        final String sql =
+                String.format(
+                        "insert into MySink SELECT "
```

Comment:
```suggestion
                        "INSERT INTO MySink SELECT "
```
nit: for readability it's better to use uppercase for keywords

Reply:
Ah, good catch.  Updated. 

FWIW, this file is just moving around.